core/object/object.cpp

```diff
@@ -892,7 +892,7 @@ void Object::set_meta(const StringName &p_name, const Variant &p_value) {
 	if (E) {
 		E->value = p_value;
 	} else {
-		ERR_FAIL_COND(!p_name.operator String().is_valid_identifier());
+		ERR_FAIL_COND_MSG(!p_name.operator String().is_valid_identifier(), "Invalid identifier");
```

Comment:
Probably makes sense to also tell what was that invalid identifier:
```suggestion
		ERR_FAIL_COND_MSG(!p_name.operator String().is_valid_identifier(), "Invalid identifier: '" + p_name + "'.");
```